doc/services/crypto/psa_crypto.rst

```diff
@@ -0,0 +1,181 @@
+
+Algorithm Flexibility
+  PSA Crypto API supports a wide range of cryptographic algorithms,
+  allowing developers to switch between different cryptographic
+  methods as needed. This flexibility is crucial for maintaining
+  security as new algorithms emerge and existing ones become obsolete.
+
+Key Management
+  The PSA Crypto API includes robust key management features that
+  support the creation, storage, and use of cryptographic keys in a
+  secure and flexible manner. It uses opaque key identifiers, which
+  allows for easy key replacement and updates without exposing key
+  material.
+
+Implementation Independence
+  The PSA Crypto API abstracts the underlying cryptographic library,
+  meaning that the specific implementation can be changed without
+  affecting the application code. This abstraction supports
+  cryptographic agility by enabling the use of different cryptographic
+  libraries or hardware accelerators as needed.
+
+Future-Proofing
+  By adhering to cryptographic agility, PSA Crypto ensures that
+  applications can quickly adapt to new cryptographic standards and
+  practices, enhancing long-term security and compliance.
+
+Examples of Applications
+************************
+
+Network Security (TLS)
+  The API provides all of the cryptographic primitives needed to establish TLS connections.
+
+Secure Storage
+  The API provides all primitives related to storage encryption, block
+  or file-based, with master encryption keys stored inside a key store.
+
+Network Credentials
+  The API provides network credential management inside a key store,
+  for example, for X.509-based authentication or pre-shared keys on
+  enterprise networks.
+
+Device Pairing
+  The API provides support for key agreement protocols that are often
+  used for secure pairing of devices over wireless channels. For
+  example, the pairing of an NFC token or a Bluetooth device might use
+  key agreement protocols upon first use.
+
+Secure Boot
+  The API provides primitives for use during firmware integrity and
+  authenticity validation, during a secure or trusted boot process.
+
+Attestation
+  The API provides primitives used in attestation
+  activities. Attestation is the ability for a device to sign an array
+  of bytes with a device private key and return the result to the
+  caller. There are several use cases; ranging from attestation of the
+  device state, to the ability to generate a key pair and prove that it
+  has been generated inside a secure key store. The API provides access
+  to the algorithms commonly used for attestation.
+
+Factory Provisioning
+  Most IoT devices receive a unique identity during the factory
+  provisioning process, or once they have been deployed to the
+  field. This API provides the APIs necessary for populating a device
+  with keys that represent that identity.
+
+Usage considerations
+********************
+
+Always check for errors
+  Most functions in the Crypto API can return errors. All functions
```

Comment:
This is the only occurrence of `Crypto API` that doesn't include `PSA`, not sure whether you want to add it here as well.